projects/dsp-ui/src/lib/viewer/services/upload-file.service.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { AppInitService } from '../../core/app-init.service';
+import { SessionService } from '../../core/session.service';
+
+interface UploadedFile {
+    fileType: string;
+    internalFilename: string;
+    originalFilename: string;
+    temporaryUrl: string;
+}
+
+export interface UploadedFileResponse {
+    uploadedFiles: UploadedFile[];
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class UploadFileService {
+
+    envUrl: string = this._is.config[`sipiUrl`];
+
+    constructor(
+        private readonly _is: AppInitService,
+        private readonly _http: HttpClient,
+        private readonly _ss: SessionService
+    ) { }
+
+    upload(file: FormData): Observable<UploadedFileResponse> {
+        const baseUrl = `${this.envUrl}upload`;
+        console.log('SESSSION', this._ss.getSession());
+        const jwt = this._ss.getSession().user.jwt;
```

Comment:
@tobiasschweizer good point, have never tested it while not logged in. I'm not sure how this component will be used and on which level it should be prevented. I believe whole form, which this input will be part of, will be just unavailable for not logged users? @kilchenmann cod you confirm it please?

Reply:
The most important thin is to perform a check if session is `null` before accessing `user`. This can then also be tested in the service's spec by passing in a session that is `null`. Let me know if can help with that.

Reply:
Sure this should be easy. I can try also to add login check before upload. Let me see.

Reply:
Fixed with 679fc9b.

Reply:
Thanks!

This was adding the `?`, right? `this._ss.getSession()?.user.jwt;`

Reply:
Correct